IPython/core/interactiveshell.py

```diff
@@ -166,6 +166,10 @@ def raise_error(self):
         if self.error_in_exec is not None:
             raise self.error_in_exec
 
+    def __repr__(self):
+        return '<%s object at %x, execution_count=%s error_before_exec=%s error_in_exec=%s result=%s>' %\
+                (self.__class__.__qualname__, id(self), self.execution_count, self.error_before_exec, self.error_in_exec, repr(self.result))
```

Comment:
If you do `%r` (or `{!r}` in new-style formatting), it repr-s the object for you, rather than needing an explicit repr() call. This isn't important, however.